dotCMS/src/main/java/com/dotcms/rest/MapToContentletPopulator.java

```diff
@@ -349,32 +386,46 @@ public List<Category> getCategories (final Contentlet contentlet, final User use
      * @param respectFrontendRoles
      * @return
      */
-    List<Category> categories(final Contentlet contentlet, final User user, final boolean respectFrontendRoles) {
-        List<Category> categories = new ArrayList<>();
-        final List<com.dotcms.contenttype.model.field.Field> fields = contentlet.getContentType()
-                .fields(CategoryField.class);
-        for (final com.dotcms.contenttype.model.field.Field field: fields) {
+    Optional<List<Category>> internalFetchCategories(final Contentlet contentlet, final User user,
```

Comment:
since both methods uses the Close db you can just resume the close db in this private method

Reply:
true. But .. please note that the old method `getCategories` is getting deprecated in favor of a new on.
done